web/html/src/manager/virtualization/guests/list/MigrateDialog.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from "react";
+import _isNil from "lodash/isNil";
+import { DangerDialog } from "components/dialog/DangerDialog";
+import { Form } from "components/input/Form";
+import { Select } from "components/input/Select";
+
+type Props = {
+  /** The modal dialog id */
+  id: string,
+  /** Virtual Machine descriptor as the one used in the VMs list data */
+  vm: any,
+  onConfirm: (vm: any, target: string) => any,
+  onClose?: () => any,
+  clusterNodes?: string[],
+};
+
+/**
+ * A pop-up dialog for virtual machine migration
+ * It contains controls to select the target virtualization host,
+ * a 'Cancel' button and a 'Migrate' button.
+ * Related virtual machine descriptor object may be passed with the 'vm' property.
+ * This 'vm' will be passed to the 'onConfirm' and 'onClosePopUp' handlers.
+ */
+export function MigrateDialog(props: Props) {
+  const [model, setModel] = React.useState<{target: string | undefined}>({target: undefined});
+  const [valid, setValid] = React.useState(false);
+  const onSubmit = () => {
+    if (!_isNil(model.target)) {
+      props.onConfirm(props.vm, model.target);
+    }
+  };
+
+  // FIXME Ideally we would need to set the AppElement as document here:
```

Comment:
What's the reason not to do this right now?

Reply:
that comment is an old one and needs to be removed: this is already done in `Dialog.tsx`